source/common/http/async_client_impl.h

```diff
@@ -216,11 +218,19 @@ class AsyncStreamImpl : public AsyncClient::Stream,
     RouteEntryImpl(
         const std::string& cluster_name, const absl::optional<std::chrono::milliseconds>& timeout,
         const Protobuf::RepeatedPtrField<envoy::config::route::v3::RouteAction::HashPolicy>&
-            hash_policy)
+            hash_policy,
+        const absl::optional<envoy::config::route::v3::RetryPolicy>& retry_policy)
         : cluster_name_(cluster_name), timeout_(timeout) {
       if (!hash_policy.empty()) {
         hash_policy_ = std::make_unique<HashPolicyImpl>(hash_policy);
       }
+      if (retry_policy.has_value()) {
+        // ProtobufMessage::getStrictValidationVisitor() ?  how often do we do this?
+        retry_policy_ = std::make_unique<Router::RetryPolicyImpl>(
+            retry_policy.value(), ProtobufMessage::getNullValidationVisitor());
+      } else {
+        retry_policy_.reset(new NullRetryPolicy());
```

Comment:
nit:
```suggestion
        retry_policy_ = std::make_unique<NullRetryPolicy>();
```

Reply:
done. 
can you please weigh in on the comment above ?
the pre-submit azure pipeline still fails, most likely because i force pushed in the same branch the re-implementation, and the ci scripts do not like it.

should I just take the same code and create a new PR from scratch to make it automatically rebase correctly and pass the builds ?